dockstore-webservice/src/main/java/io/dockstore/webservice/resources/AbstractWorkflowResource.java

```diff
@@ -654,19 +667,22 @@ private boolean createWorkflowsAndVersionsFromDockstoreYml(List<? extends Workfl
                 try {
                     DockstoreYamlHelper.validate(wf, true, "a " + computeTermFromClass(workflowType));
 
+                    // Retrieve the user who triggered the call (must exist on Dockstore if workflow is not already present)
+                    User user = GitHubHelper.findUserByGitHubUsername(this.tokenDAO, this.userDAO, username, false);
+
                     // Update the workflow version in its own database transaction.
+                    AtomicLong workflowId = new AtomicLong();
+                    AtomicLong workflowVersionId = new AtomicLong();
                     transactionHelper.transaction(() -> {
                         final String workflowName = workflowType == Service.class ? "" : wf.getName();
                         final Boolean publish = wf.getPublish();
                         final boolean latestTagAsDefault = wf.getLatestTagAsDefault();
                         final List<YamlAuthor> yamlAuthors = wf.getAuthors();
 
-                        // Retrieve the user who triggered the call (must exist on Dockstore if workflow is not already present)
-                        User user = GitHubHelper.findUserByGitHubUsername(this.tokenDAO, this.userDAO, username, false);
-
                         Workflow workflow = createOrGetWorkflow(workflowType, repository, user, workflowName, wf, gitHubSourceCodeRepo);
+                        workflowId.set(workflow.getId());
                         WorkflowVersion version = addDockstoreYmlVersionToWorkflow(repository, gitReference, dockstoreYml, gitHubSourceCodeRepo, workflow, latestTagAsDefault, yamlAuthors);
-
+                        workflowVersionId.set(version.getId());
```

Comment:
As part of https://ucsc-cgl.atlassian.net/browse/SEAB-6449, I can modify `TransactionHelper` to include a method that allows us to run a method in a transaction that returns a result.  Then, we don't need to track the ids with the AtomicLongs if we opt not to.